From 35403aaf647e6f4ed8ae31f19a61c3704458443f Mon Sep 17 00:00:00 2001
From: Richard Zhu <hongxing.zhu@nxp.com>
Date: Tue, 28 Feb 2017 09:22:23 +0800
Subject: [PATCH 1549/5242] MLK-14325-1 ARM: dts: enable imx pcie gen2 link

commit  5b2247605757852e725165b73bc842bcc0dce463 from
https://source.codeaurora.org/external/imx/linux-imx.git

Set the max link capability of the imx pcie to gen2

Signed-off-by: Richard Zhu <hongxing.zhu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx6qdl.dtsi       |    1 +
 arch/arm/boot/dts/imx6qp-sabresd.dts |    4 ----
 arch/arm/boot/dts/imx6sx.dtsi        |    1 +
 3 files changed, 2 insertions(+), 4 deletions(-)

diff --git a/arch/arm/boot/dts/imx6qdl.dtsi b/arch/arm/boot/dts/imx6qdl.dtsi
index adabb36..4c04af3 100644
--- a/arch/arm/boot/dts/imx6qdl.dtsi
+++ b/arch/arm/boot/dts/imx6qdl.dtsi
@@ -307,6 +307,7 @@
 				 <&clks IMX6QDL_CLK_LVDS1_GATE>,
 				 <&clks IMX6QDL_CLK_PCIE_REF_125M>;
 			clock-names = "pcie", "pcie_bus", "pcie_phy";
+			fsl,max-link-speed = <2>;
 			status = "disabled";
 		};
 
diff --git a/arch/arm/boot/dts/imx6qp-sabresd.dts b/arch/arm/boot/dts/imx6qp-sabresd.dts
index ddd9b68..c81f63e3 100644
--- a/arch/arm/boot/dts/imx6qp-sabresd.dts
+++ b/arch/arm/boot/dts/imx6qp-sabresd.dts
@@ -50,10 +50,6 @@
 	};
 };
 
-&pcie {
-	status = "disabled";
-};
-
 &iomuxc {
 	imx6qdl-sabresd {
 		pinctrl_usdhc2: usdhc2grp {
diff --git a/arch/arm/boot/dts/imx6sx.dtsi b/arch/arm/boot/dts/imx6sx.dtsi
index d53b4f0..047d1b8 100644
--- a/arch/arm/boot/dts/imx6sx.dtsi
+++ b/arch/arm/boot/dts/imx6sx.dtsi
@@ -1573,6 +1573,7 @@
 			clock-names = "pcie", "pcie_bus", "pcie_phy", "pcie_inbound_axi";
 			pcie-phy-supply = <&reg_pcie_phy>;
 			power-domains = <&pd_pci>;
+			fsl,max-link-speed = <2>;
 			status = "disabled";
 		};
 	};
-- 
1.7.9.5

